src/scry/analyzer.cr

```diff
@@ -5,12 +5,22 @@ require "./publish_diagnostic"
 
 module Scry
   struct Analyzer
+    @@source = [] of String
```

Comment:
In VSCode, every workspace launch a new instance of scry, in NVIM too. I don't know in Atom, however if the code inside `text_document.text` changes, then the source file is updated.

My goals with this variable is to check that source is different when document is saved, because the actual behavior is checking the file too much, spending memory and CPU time.

Is a class variable because a new instance of Analyzer is created  in a new message, and I didn't have other way to handle this.